### PR TITLE
Contributor area: Improve readme + feature policy

### PR DESCRIPTION
**Pull request by SgtCoDFish**
As discussed in retro today, this PR adds more detail of a feature policy in an effort to make it easier for both contributors and maintainers alike.

Also reworks the README for the contributing section too, and changes a bunch of language.

### CODE REVIEW

File: content/docs/contributing/policy.md
Comment (irbekrm) [inline content/docs/contributing/policy.md:31]: Question: does this part essentially suggest that a design should be PRed to https://github.com/cert-manager/cert-manager/tree/master/design? (Which I thought was what we wanted)

File: content/docs/contributing/policy.md
Comment (irbekrm) [inline content/docs/contributing/policy.md:34]: Thought: upstream SIGs seem to discuss designs in their dev meetings and it looks like usually a person can just add a design to be discussed to the meeting notes (seems like usually after having had the PR with the design open for a little while to give maintainers a chance to take a look). Would it make sense for us to also do that?

File: content/docs/contributing/policy.md
Comment (irbekrm) [reply content/docs/contributing/policy.md:31]: I guess this is also where the design doc template would come in as there would be some generic things that all docs such docs should include (i.e whether this change involves an API change, whether it will be feature gated etc)

File: content/docs/contributing/policy.md
Comment (irbekrm) [inline content/docs/contributing/policy.md:14]: Absolute nit: I guess it is also whether the change that they involve is 'significant' or not

File: content/docs/contributing/policy.md
Comment (SgtCoDFish) [reply content/docs/contributing/policy.md:31]: It doesn't suggest that - it just says "raise an issue". I think PRing to the design folder might be a better shout, though.

I basically just suggested the thing that was the least friction for people here. But having this stuff versioned and reviewed through a PR is probably preferable! I'll update.

**Review by irbekrm — COMMENTED** [review]
Thanks for working on this!

It looks like a great improvement to me. I've added a few questions, but generally would be happy to lgtm this after others have had a chance to take a look

File: content/docs/contributing/policy.md
Comment (SgtCoDFish) [reply content/docs/contributing/policy.md:14]: Yeah, that's worth saying explicitly, thanks!

File: content/docs/contributing/policy.md
Comment (SgtCoDFish) [reply content/docs/contributing/policy.md:34]: Yeah I'll call adding things to meeting notes as an option explicitly; nice suggestion, thanks :+1:

**Comment by SgtCoDFish** [comment]
@irbekrm I've updated the language a bit and split the "larger features" bit into sections. Also added a little timeline!

**Comment by irbekrm** [comment]
Thanks for updating it!

This makes sense to me- adding an lgtm with a hold to allow others to take a look- feel free to unhold when you wish

/lgtm
/hold

**Comment by SgtCoDFish** [comment]
(NB: I didn't change any content, I rebased against latest master)

File: content/docs/contributing/policy.md
Comment (jakexks) [inline content/docs/contributing/policy.md:6]:
```suggestion
We love to receive both feature requests and PRs which improve to cert-manager; the community is at the heart of what we do!
```

I don't like the phrase "add stuff" (but also don't mind if it remains that much).

File: content/docs/contributing/policy.md
Comment (jakexks) [inline content/docs/contributing/policy.md:10]:
```suggestion
We recommend creating an issue first for it to be discussed with the cert-manager maintainers. Another possibility is bringing it up in a community meeting for an open discussion with the maintainers and community on the implementation.
```

Repeating the word discussion so close together seems clumsy. Maybe repeating community is also unnecessary?

**Review by jakexks — APPROVED** [review]
This seems a good improvement! I have a couple of nitpicks but nothing major.
/lgtm
/hold

File: content/docs/contributing/policy.md
Comment (SgtCoDFish) [reply content/docs/contributing/policy.md:6]: Fair enough, I get it; will change :+1:

File: content/docs/contributing/policy.md
Comment (SgtCoDFish) [reply content/docs/contributing/policy.md]: Yeah this was definitely an editing mistake on my part, good catch :+1: Fixed!

**Comment by SgtCoDFish** [comment]
@irbekrm  @jakexks  I've addressed all comments now, please feel free to take another look :grin:

**Pull request merged**
